jep/0000/README.adoc

```diff
@@ -0,0 +1,309 @@
+    - `jenkins.security.s2m.FilePathRuleConfig` is deleted.
+    - `jenkins.security.s2m.MasterKillSwitchConfiguration` is deleted, including its resources.
+    - `jenkins.security.s2m.MasterKillSwitchWarning` is deleted, including its resources.
+    - `jenkins.security.s2m.OpMatcher` is deleted.
+    - `jenkins.security.s2m.RejectedCallable` is deleted.
+    - `jenkins.security.s2m.RunningBuildFilePathFilter` is deleted.
+    - `resources/jenkins/security/s2m/callable.conf` is deleted.
+    - `resources/jenkins/security/s2m/filepath-filter.conf` is deleted.
+* `AdminWhitelistRule` is deprecated, its resources deleted, and all functionality removed:
+    - The methods `#setMasterKillSwitch(boolean)` and `#getMasterKillSwitch()` are changed so they only log messages informing about its new lack of functionality.
+    - Everything else is removed.
+* `FilePath` is updated to no longer use any `FilePathFilter` functionality:
+    - The private `SecureFileCallable` (the marker interface for `SlaveToMasterFileCallable` using `FilePathFilter`) is removed and every `FileCallable` in `FilePath` now extends `MasterToSlaveFileCallable` instead.
+    - All private static functions like `#reading(File)` that perform access checks are removed, as well as `#filterNonNull()` which supported them.
+* `jenkins.security.s2m.CallableDirectionChecker` is simplified and now always enforces role checks and disables class loading from agents.
+* `SetupWizard` no longer sets the `AdminWhitelistRule` kill switch, as protections are now always effective.
+* The `jenkins.security.s2m.CallableDirectionChecker.allow` system property escape hatch is retained:
+    - It allows classloading from agents to the controller (as before).
+    - It allows executing any callable regardless of its role check (as before).
+    - All `FileCallable` implementations, including those in `FilePath`, are allowed to act on any path.
+
+== Motivation
+
+The agent-to-controller security subsystem was added in 2014 to restrict the actions that agent processes can perform on the Jenkins controller as part of https://www.jenkins.io/security/advisory/2014-10-30/[the SECURITY-144 security fix].
+This protection was comprised of three major, complementary parts:
+
+Disabled class loading::
+Controllers do not load classes from agents, which means all code on a controller must already be part of that environment.
+No new code can be injected from agents.
+Role checks::
+Every `Callable` declares through its role check whether it's allowed to be sent from an agent to a controller.
+Legacy callables (built for Jenkins 1.565.3 or older, or Jenkins 1.586 or older) were rejected by default, but admins could allow their transmission from agents to the controller.
+File path filters::
+To continue supporting various methods on `FilePath` that transparently access files on the other side of a remoting channel, file path filters limit which files and directories can be accessed through them in what manner.
```

Comment:
```suggestion
To continue supporting various methods on `FilePath` that transparently access files on the other side of a remoting channel, file path filters limit which files and directories can be accessed through them in whatever manner.
```

Reply:
Not entirely sure what I meant, but I think it was
```suggestion
To continue supporting various methods on `FilePath` that transparently access files on the other side of a remoting channel, file path filters limit which files and directories can be accessed through them in that manner.
```

Reply:
```suggestion
To continue supporting various methods on `FilePath` that transparently access files on the other side of a remoting channel, file path filters limit which files and directories can be accessed.
```

would suffice I think?